src/main/java/com/google/api/gax/grpc/PageStreamingCallable.java

```diff
@@ -60,27 +62,39 @@ public String toString() {
     return String.format("pageStreaming(%s)", callable);
   }
 
-  public ListenableFuture<Iterable<ResourceT>> futureCall(CallContext<RequestT> context) {
-    return Futures.immediateFuture((Iterable<ResourceT>)new StreamingIterable(context));
+  @Override
+  public ListenableFuture<PageAccessor<ResourceT>> futureCall(CallContext<RequestT> context) {
+    PageAccessor<ResourceT> pageAccessor =
+        new PageAccessorImpl<RequestT, ResponseT, ResourceT>(callable, pageDescriptor, context);
+    return Futures.immediateFuture(pageAccessor);
   }
 
-  private class StreamingIterable implements Iterable<ResourceT> {
+  private class PageAccessorImpl<RequestT, ResponseT, ResourceT>
+      implements PageAccessor<ResourceT> {
+    private final FutureCallable<RequestT, ResponseT> callable;
+    private final PageStreamingDescriptor<RequestT, ResponseT, ResourceT> pageDescriptor;
     private final CallContext<RequestT> context;
+    private ResponseT currentPage;
```

Comment:
Because currentPage is not null I think this implementation is not thread-safe (concurrent calls of `getPageValues() `
or `getNextPageToken()` may get different or not consistent values). This can be fine if desired *and* documented.